### PR TITLE
Fix assignment to heterogeneously stored tuples

### LINKED ISSUE #1160: Returning multiple values in a function
### Version Information

* vyper Version: 0.1.0
* OS: 18.04

### What's your issue about?
Unable to set tuples aka #605 

Please include information like:

* full output of the error you received
test_a remains as 100

* what command you ran
checking events emitted using truffle
truffle test

* the code that caused the failure 
```
Creation: event({
    _test_a: uint256
})
test_a : uint256
test_b : uint256
test_c : uint256
test_x : uint256


@public
def __init__():
    self.test_a = 100
    
@public
def test1() -> (uint256, uint256):
    log.Creation(self.test_x)
    return (1,2)

@public
def test():
    (self.test_b, self.test_a) = self.test1()
    log.Creation(self.test_a)
    log.Creation(self.test_b)
```

Test Code:
```
const test = artifacts.require("test");
const assert = require("chai").assert;
const truffleAssert = require('truffle-assertions');

contract("test", (accounts) => {
    let test1;
    const fundingAccount = accounts[0];

    beforeEach(async () => {
        test1 = await test.new({from: fundingAccount});
    });
    
    it("should receive event", async() => {
        let tx = await test1.test();
        truffleAssert.eventEmitted(tx, 'Creation', (ev) => {
            return ev._test_a == 2;
        });
    });
});
```

I am not sure whether I did it wrong but the goal would be to set 2 variables at once.

#### Cute Animal Picture
image

### PR DESCRIPTION
**Pull request by charles-cooper**
### - What I did
Fix #1160

### - How I did it
Set location correctly when the tuple elements have different locations (as can happen when the left hand is a tuple literal).

### - How to verify it
Tests pass
Also it can be manually verified by compiling the contract in #1160 and verifying that the `sstore`s are generated in the right places.

### CODE REVIEW

**Comment by charles-cooper** [comment]
Not sure why the CI is failing .. flake8 passes on my machine.

**Comment by fubuloubu** [comment]
It looks like something else got merged that broke flake8

**Comment by charles-cooper** [comment]
Actually I think Travis is pulling a newer version of flake8. After I upgraded my local version of flake8 I can reproduce the errors.

File: vyper/parser/parser_utils.py [hunk 1]
```diff
@@ -449,26 +450,33 @@ def make_setter(left, right, location, pos, in_function_call=False):
...
+                subs.append(make_setter(add_variable_offset(left_token, typ, pos=pos), LLLnode.from_list(None, typ=NullType()), loc, pos=pos))
             return LLLnode.from_list(['with', '_L', left, ['seq'] + subs], typ=None)
         # If tuple assign.
         elif isinstance(left.typ, TupleType) and isinstance(right.typ, TupleType):
             right_token = LLLnode.from_list('_R', typ=right.typ, location="memory")
             subs = []
             static_offset_counter = 0
-            for idx, (left_arg, right_arg) in enumerate(zip(left.args, right.typ.members)):
+            for idx, (left_arg, right_arg, loc) in enumerate(zip(left.args, right.typ.members, locations)):
                 # if left_arg.typ.typ != right_arg.typ:
```
Comment (jacqueswww) [inline vyper/parser/parser_utils.py]: I believe this test can be removed? Do we have a test that checks types are correct?

Comment (charles-cooper) [reply vyper/parser/parser_utils.py]: Yep! The type check is implicitly done in the recursive call to `make_setter`.

Comment (jacqueswww) [reply vyper/parser/parser_utils.py]: Thought so :) just drop the comment and merge (sorry was a bit eager on the merge).

**Pull request merged**